### PR TITLE
updated physics/archimedes_principle.py

### PR DESCRIPTION
**Pull request by Baron105**
### Describe your change:

modified physics/archimedes_principle.py
added some doctests for exception handling, imported g from scipy rather than hard coding and allowed zero gravity to be valid giving value 0 in that case

fixes issue #9943 

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: physics/archimedes_principle.py [hunk 1]
```diff
@@ -10,8 +10,8 @@
 """
 
 
-# Acceleration Constant on Earth (unit m/s^2)
-g = 9.80665
+# Acceleration Constant on Earth (unit m/s^2) imported from scipy
+from scipy.constants import g
```
Comment (cclauss) [inline physics/archimedes_principle.py:14]: Please revert this change.  Let's not create a dependency on all of scipy just to set a constant.

File: physics/archimedes_principle.py [hunk 2]
```diff
@@ -30,13 +30,27 @@ def archimedes_principle(
...
+    ValueError: Impossible Gravity
     """
 
     if fluid_density <= 0:
         raise ValueError("Impossible fluid density")
-    if volume < 0:
+    if volume <= 0:
         raise ValueError("Impossible Object volume")
-    if gravity <= 0:
+    if gravity < 0:
```
Comment (cclauss) [inline physics/archimedes_principle.py:53]: Please revert these changes.  It is useful to be able to make theoretical calculations even if we are unable to reproduce those conditions.

**Review by cclauss — CHANGES_REQUESTED** [review]

File: physics/archimedes_principle.py (hunk 2 above)
Comment (Baron105) [reply physics/archimedes_principle.py:53]: should I remove the tests as well or just make the signs as they were before?

Comment (cclauss) [reply physics/archimedes_principle.py:53]: The tests need to follow the code.  It is OK to have the tests but their output must pass.

Comment (Baron105) [reply physics/archimedes_principle.py:53]: The numerical tests are passing. Just added some more tests to see for when the inputs are invalid physically as in negative gravity, non positive volume and non positive density as they were already in the original code by the author.

Comment (Baron105) [reply physics/archimedes_principle.py:53]: please check @cclauss

Comment (cclauss) [reply physics/archimedes_principle.py:53]: Dude!  There are 215 pull requests open.  Perhaps we will get to review this one.

**Review by cclauss — APPROVED** [review]
Nice!!

**Comment by Baron105** [comment]
Great! Looks better now!
